ldap/src/main/java/org/springframework/security/ldap/DefaultSpringSecurityContextSource.java

```diff
@@ -136,7 +145,13 @@ public DefaultSpringSecurityContextSource(List<String> urls, String baseDn) {
 	private static String buildProviderUrl(List<String> urls, String baseDn) {
 		Assert.notNull(baseDn, "The Base DN for the LDAP server must not be null.");
 		Assert.notEmpty(urls, "At least one LDAP server URL must be provided.");
-		String trimmedBaseDn = baseDn.trim();
+		String trimmedAndEncodedBaseDn;
+		try {
+			trimmedAndEncodedBaseDn = URLEncoder.encode(baseDn.trim(), StandardCharsets.UTF_8.toString());
```

Comment:
For added readability, consider moving the exception handling out to a separate method.